### DISCUSSION TITLE
Rsi in Double data type !

### DISCUSSION THREAD

**Discussion by moslem7026** [DaveSkender/Stock.Indicators#928] [Q&A]
i see that data type of rsi value has been changed to double , whats going on sma that has decimal point ?
https://github.com/DaveSkender/Stock.Indicators/blob/e64d14e19d8bd0351cc2a85d453a51c52c32e226/src/m-r/Rsi/Rsi.Models.cs#L11

**Comment by DaveSkender** [comment]
We've had a rather long [prior conversation on using decimal vs. double](https://github.com/DaveSkender/Stock.Indicators/discussions/649).  I'm sure there are a few inconsistencies in the library, but we prefer `double` for performance, but persist `decimal` for precision when appropriate.

[The convention we use](https://github.com/DaveSkender/Stock.Indicators/discussions/649#discussioncomment-1792553) prefers `double` when the results are abstracted away from price to favor its relative computational performance.  For example, `RSI` is on a 0-100 point scale with low or no benefit to decimalized precision, so `double` works fine.  In price overlays or transforms, like `SMA` or `Renko`, we stuck with `decimal` to keep the price-domain usable for monetary calculations.

In version 2.0, we went further, to make all but pure price-based transforms results `double`.  Currently, we're only using `decimal` outputs for more pure candle transforms like `CandleProperties`, `Renko`, `Heikin-Ashi`, etc.  There may be a few exceptions.

**Comment by DaveSkender** [comment]
If your question is "why isn't RSI an integer?", I'd say because it's not accurate enough as an integer, especially if re-used in chaining models.  If you want it as an integer for your use case, you can convert it:

```csharp
// example, make RSI an integer
var results = quotes
    .GetRsi(..)
    .Select(x => new { x.Date, Rsi = (int)x.Rsi });
```
